src/ocnicepost.fd/ocnicepost.F90

```diff
@@ -0,0 +1,577 @@
+program ocnicepost
+
+  use utils_mod, only : debug, logunit, getfield, packarrays, remap, dumpnc
+  use outputvars
+  use netcdf
+
+  implicit none
+
+  character(len= 20) :: fnml, ftype
+  character(len=120) :: wgtsdir
+  character(len=120) :: input_file, wgtsfile, output_file
+
+  ! source grid, tripole 1/4 deg, 40 vertical levels
+  integer, parameter :: nxt = 1440, nyt = 1080, nlevs = 40
+
+  ! destination grids
+  integer, parameter :: ndest = 3
+  integer, parameter, dimension(ndest) :: nxrs = (/1440, 720, 360/)
+  integer, parameter, dimension(ndest) :: nyrs = (/ 721, 361, 181/)
+  character(len=4), dimension(ndest)   :: dstgrds = (/'0p25', '0p5 ', '1p0 '/)
+
+  ! packed arrays on source grid
+  real, allocatable, dimension(:,:)   :: bilin2d, consd2d
+  real, allocatable, dimension(:,:,:) :: bilin3d
+  ! variable types
+  type(vardefs), allocatable :: b2d(:)
+  type(vardefs), allocatable :: c2d(:)
+  type(vardefs), allocatable :: b3d(:)
+
+  ! source grid fields
+  real, dimension(nxt,nyt,nlevs) :: tmp3d
+  real, dimension(nxt*nyt,nlevs) :: mask3d
+  real, dimension(nxt*nyt)       :: mask2d
+  real, dimension(nxt*nyt)       :: cosrot, sinrot, anglet
+  real, dimension(nlevs)         :: z_l
+  real, dimension(0:nlevs)       :: z_i
+
+  ! destination grid fields
+  real, allocatable, dimension(:,:)  :: dstlon, dstlat
+  real, allocatable, dimension(:,:)  :: rgmask3d
+  real, allocatable, dimension(:)    :: rgmask2d
+  ! output fields (x,y,z)
+  real, allocatable, dimension(:)     :: out1d
+  real, allocatable, dimension(:,:)   :: out2d
+  real, allocatable, dimension(:,:,:) :: out3d
+  ! packed remapped fields
+  real, allocatable, dimension(:,:)   :: rgb2d, rgc2d
+  real, allocatable, dimension(:,:,:) :: rgb3d
+
+  real(kind=8)       :: timestamp
+  character(len= 40) :: timeunit, timecal
+  character(len= 20) :: vname, vunit
+  character(len=120) :: vlong
+  character(len=4)   :: dstgrid
+
+  real    :: vfill
+  integer :: nd, nxr, nyr
+  integer :: i,j,k,n,nn,nvalid,iounit
+  integer :: rc,ncid,varid,dimid
+  integer :: nbilin2d,nbilin3d,nconsd2d
+  integer :: idimid,jdimid,kdimid,edimid,timid
+  integer :: idx1,idx2,idx3
+  logical :: do_ocnpost
+
+  namelist /ocnicepost_nml/ ftype, wgtsdir, debug
+
+  ! --------------------------------------------------------
+  ! read the name list
+  ! --------------------------------------------------------
+
+  fnml = 'ocnicepost.nml'
```

Comment:
The namelist interface should be a separate module and subroutine to reduce the bloat of this file.